openedx/features/redhouse_panel/api/v0/serializers.py

```diff
@@ -9,7 +21,98 @@ class SiteSerializer(serializers.Serializer):
     address = serializers.CharField()
 
 
-class UserSerializer(serializers.ModelSerializer):
+class UserProfileSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = UserProfile
+        fields = ('name', 'year_of_birth')
+
+
+class UserAccountSerializer(serializers.ModelSerializer):
+    profile = UserProfileSerializer()
+    can_access_panel = serializers.SerializerMethodField('has_panel_permission')
+    is_instructor = serializers.SerializerMethodField('can_create_courses')
+
     class Meta:
         model = User
-        fields = ('id', 'first_name', 'last_name', 'is_active')
+        read_only_fields = ('id',)
+        fields = ('username', 'email', 'is_active', 'profile', 'can_access_panel', 'is_instructor')
+
+    def has_panel_permission(self, obj):
+        if obj.groups.filter(name__in=[REDHOUSE_PANEL_GROUP_NAME]).exists():
+            return True
+        return False
+
+    def can_create_courses(self, obj):
+        request = self.context['request']
+        edly_user_info_cookie = request.COOKIES.get(settings.EDLY_USER_INFO_COOKIE_NAME, None) if request else None
+        edx_org = get_edx_org_from_cookie(edly_user_info_cookie)
+
+        if auth.user_has_role(obj, GlobalCourseCreatorRole(edx_org)):
+            return True
+        return False
+
+    def create(self, validated_data):
+        # create user
+        profile_data = validated_data.pop('profile')
+        can_access_panel = self.initial_data.get('can_access_panel', None)
+        is_instructor = self.initial_data.get('is_instructor', None)
+
+        user = User.objects.create(**validated_data)
+
+        # create profile
+        UserProfile.objects.create(
+            user=user,
+            name=profile_data.get('name', None),
+            year_of_birth=profile_data.get('year_of_birth', None)
```

Comment:
there is no need to for the default value, profile_date.get('year_of_birth') will do the job

Reply:
Done